src/Extension/FluentDirectorExtension.php

```diff
@@ -159,4 +155,21 @@ protected function getRuleController($existingRule, $localeObj)
         $this->owner->extend('updateLocalePageController', $controller, $localeObj);
         return $controller;
     }
+
+    /**
+     * Inserts the given rule(s) before another rule
+     * @param array $rules Array of rules to insert before
+     * @param string $key Rule to insert the new rules before
+     * @param array $rule New Rules to insert
+     * @return array Resulting array of rules
+     */
+    protected function insertRuleBefore(array $rules, $key, array $rule)
+    {
+        $i = array_search($key, array_keys($rules));
+        if ($i !== false) {
+            return array_slice($rules, 0, $i, true) + $rule + array_slice($rules, $i, null, true);
+        }
```

Comment:
Should we expect this function to always add the rule? The `FieldList` has `appendIfMissing` as an argument which defaults to `true`.

Reply:
Since it's looking for the rule defined by the [cms module](https://github.com/silverstripe/silverstripe-cms/blob/4.3/_config/routes.yml#L22) it should based on the current module requirements. That said I could add the appendIfMissing idea, I do like that as a fall back for sure though I might do a prepend so it works essentially like Fluent currently does.